Certain tests still fail randomly in CI

Until it is understood why certain tests fail have random failures in CI, mark them with `@pytest.mark.local`.

_Originally posted by @prjemian in https://github.com/BCDA-APS/apstools/issues/747#issuecomment-1287486362_

- #696

Again!  Different place.

```py
        assert round(pos.readback.get(use_monitor=False), 2) == known_start_position
>       assert pos.inposition, f"target={pos.setpoint.get()}, readback={pos.position}"
E       AssertionError: target=-1.0, readback=-1.0
E       assert False
E        +  where False = PVPositionerSoftDoneWithStop(prefix='gp:gp:', name='pos', settle_time=0.0, timeout=None, read_attrs=['readback', 'setpoint'], configuration_attrs=['done', 'tolerance', 'target'], limits=None, egu='').inposition

/home/runner/work/apstools/apstools/apstools/devices/tests/test_positioner_soft_done.py:321: AssertionError
```

Another:

```py
>       assert math.isclose(rb, sp, abs_tol=positioner.actual_tolerance)
E       AssertionError: assert False
E        +  where False = <built-in function isclose>(0.0, -1.0, abs_tol=1e-05)
E        +    where <built-in function isclose> = math.isclose
E        +    and   1e-05 = PVPositionerSoftDoneWithStop(prefix='', name='calcpos', settle_time=0.0, timeout=None, read_attrs=['readback', 'setpoint'], configuration_attrs=['done', 'tolerance', 'target'], limits=None, egu='').actual_tolerance

/home/runner/work/apstools/apstools/apstools/devices/tests/test_positioner_soft_done.py:275: AssertionError
```

--- a/.github/workflows/linter.yml
+++ b/.github/workflows/linter.yml
@@ -44,7 +44,7 @@ jobs:
       # Run Linter against code base #
       ################################
       - name: Lint Code Base
-        uses: github/super-linter@v4
+        uses: github/super-linter/slim@v4
         env:
           VALIDATE_ALL_CODEBASE: false
           VALIDATE_JSCPD: false

--- a/apstools/callbacks/tests/test_i806_nxwriter.py
+++ b/apstools/callbacks/tests/test_i806_nxwriter.py
@@ -81,7 +81,7 @@ def test_as_reported():
     noisy = EpicsSignalRO(NOISY_PV, name="noisy")
     user_transforms = UserTransformsDevice(IOC, name="user_transforms")
     for obj in (motor, noisy, und, user_transforms):
-        obj.wait_for_connection()
+        obj.wait_for_connection(timeout=20.0)
 
     assert und.name == "und"
     assert und.name in und.read()

--- a/apstools/devices/tests/test_positioner_soft_done.py
+++ b/apstools/devices/tests/test_positioner_soft_done.py
@@ -206,7 +206,7 @@ def test_structure(device, has_inposition):
     assert pos.tolerance.get() == -1
 
 
-@pytest.mark.local
+# @pytest.mark.local
 def test_put_and_stop(rbv, prec, pos):
     assert pos.tolerance.get() == -1
     assert pos.precision == prec.get()
@@ -216,7 +216,7 @@ def test_put_and_stop(rbv, prec, pos):
         c_sp = pos._sp_count
         pos.setpoint.put(target)
         assert pos._sp_count == c_sp + 1
-        assert math.isclose(pos.readback.get(), rb_initial, abs_tol=0.02)
+        assert math.isclose(pos.readback.get(use_monitor=False), rb_initial, abs_tol=0.02)
         assert math.isclose(pos.setpoint.get(use_monitor=False), target, abs_tol=0.02)
         assert pos.done.get() != pos.done_value
         assert not pos.inposition
@@ -247,7 +247,7 @@ def test_put_and_stop(rbv, prec, pos):
     motion(1, 0, 0.5)  # interrupted move
 
 
-@pytest.mark.local
+# @pytest.mark.local
 def test_move_and_stop_nonzero(rbv, pos):
     timed_pause()
 
@@ -264,6 +264,10 @@ def test_move_and_stop_nonzero(rbv, pos):
     timed_pause()
     # assert dt >= longer_delay
     assert pos.inposition
+
+    rb_new = pos.readback.get(use_monitor=False)
+    arrived = math.isclose(rb_new, target, abs_tol=pos.actual_tolerance)
+    assert arrived, f"{target=}"
 
 
 # @pytest.mark.local

--- a/apstools/devices/tests/test_positioner_soft_done.py
+++ b/apstools/devices/tests/test_positioner_soft_done.py
@@ -206,6 +206,7 @@ def test_structure(device, has_inposition):
     assert pos.tolerance.get() == -1
 
 
+@pytest.mark.local
 def test_put_and_stop(rbv, prec, pos):
     assert pos.tolerance.get() == -1
     assert pos.precision == prec.get()
@@ -247,6 +248,7 @@ def test_put_and_stop(rbv, prec, pos):
     motion(1, 0, 0.5)  # interrupted move
 
 
+@pytest.mark.local
 def test_move_and_stop_nonzero(rbv, pos):
     timed_pause()
 
@@ -269,6 +271,7 @@ def test_move_and_stop_nonzero(rbv, pos):
     assert pos.inposition
 
 
+@pytest.mark.local
 def test_move_and_stopped_early(rbv, pos):
     def motion(target, delay, interrupt=False):
         timed_pause(0.1)  # allow previous activities to settle down

--- a/apstools/devices/tests/test_positioner_soft_done.py
+++ b/apstools/devices/tests/test_positioner_soft_done.py
@@ -261,13 +261,13 @@ def test_move_and_stop_nonzero(rbv, pos):
     assert status.done
     assert status.success
     # assert status.elapsed >= longer_delay
-    timed_pause()
-    # assert dt >= longer_delay
-    assert pos.inposition
 
     rb_new = pos.readback.get(use_monitor=False)
     arrived = math.isclose(rb_new, target, abs_tol=pos.actual_tolerance)
     assert arrived, f"{target=}"
+
+    # assert dt >= longer_delay
+    assert pos.inposition
 
 
 # @pytest.mark.local

--- a/apstools/devices/tests/test_positioner_soft_done.py
+++ b/apstools/devices/tests/test_positioner_soft_done.py
@@ -206,7 +206,6 @@ def test_structure(device, has_inposition):
     assert pos.tolerance.get() == -1
 
 
-# @pytest.mark.local
 def test_put_and_stop(rbv, prec, pos):
     assert pos.tolerance.get() == -1
     assert pos.precision == prec.get()
@@ -248,7 +247,6 @@ def test_put_and_stop(rbv, prec, pos):
     motion(1, 0, 0.5)  # interrupted move
 
 
-# @pytest.mark.local
 def test_move_and_stop_nonzero(rbv, pos):
     timed_pause()
 
@@ -271,7 +269,6 @@ def test_move_and_stop_nonzero(rbv, pos):
     assert pos.inposition
 
 
-# @pytest.mark.local
 def test_move_and_stopped_early(rbv, pos):
     def motion(target, delay, interrupt=False):
         timed_pause(0.1)  # allow previous activities to settle down

--- a/apstools/devices/tests/test_positioner_soft_done.py
+++ b/apstools/devices/tests/test_positioner_soft_done.py
@@ -235,7 +235,8 @@ def test_put_and_stop(rbv, prec, pos):
             assert pos.position == rb_mid
         else:  # interrupted move
             rbv.put(target)  # make the readback match
-            assert math.isclose(pos.readback.get(use_monitor=False), target, abs_tol=0.02)
+            rb_new = pos.readback.get(use_monitor=False)
+            assert math.isclose(rb_new, target, abs_tol=0.02)
 
         assert pos.inposition
         assert pos.done_value is True

--- a/apstools/devices/tests/test_positioner_soft_done.py
+++ b/apstools/devices/tests/test_positioner_soft_done.py
@@ -266,7 +266,7 @@ def test_move_and_stop_nonzero(rbv, pos):
     assert pos.inposition
 
 
-@pytest.mark.local
+# @pytest.mark.local
 def test_move_and_stopped_early(rbv, pos):
     def motion(target, delay, interrupt=False):
         timed_pause(0.1)  # allow previous activities to settle down
@@ -278,9 +278,8 @@ def test_move_and_stopped_early(rbv, pos):
         timed_pause(delay / 2 if interrupt else delay + 0.1)
         dt = time.time() - t0
         # fmt: off
-        arrived = math.isclose(
-            pos.readback.get(use_monitor=False), target, abs_tol=pos.actual_tolerance
-        )
+        rb_new = pos.readback.get(use_monitor=False)
+        arrived = math.isclose(rb_new, target, abs_tol=pos.actual_tolerance)
         # fmt: on
         if interrupt:
             assert not status.done

--- a/apstools/plans/tests/test_alignment.py
+++ b/apstools/plans/tests/test_alignment.py
@@ -60,9 +60,9 @@ def change_noisy_parameters(fwhm=0.15, peak=10000, noise=0.08):
 
 def get_position(obj, digits=3):
     if isinstance(obj, ophyd.EpicsMotor):
-        position = obj.position
+        position = obj.user_readback.get(use_monitor=False)
     else:
-        position = obj.get()
+        position = obj.get(use_monitor=False)
     return round(position, digits)
 
 
@@ -127,11 +127,11 @@ def test_direct_implementation_with_rel_scan(signal, mover, start, finish, npts,
         #     center = swait.channels.B.input_value.get()
         #     width = swait.channels.C.input_value.get()
         position = get_position(mover)
-        lo = center-width
-        hi = center+width
+        lo = center - width
+        hi = center + width
         fwhm = bec.peaks["fwhm"][signal.name]
-        assert lo <= position+fwhm, f"{lo=} {position=} {fwhm=}"
-        assert position-fwhm < hi, f"{position=} {fwhm=} {hi=}"
+        assert lo <= position + fwhm, f"{lo=} {position=} {fwhm=}"
+        assert position - fwhm < hi, f"{position=} {fwhm=} {hi=}"
 
 
 @pytest.mark.parametrize(

